react-basic-practice/src/contexts/loading.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext, useMemo, useState } from 'react';
+
+const LoadingContext = React.createContext();
+
+const useLoading = () => useContext(LoadingContext);
+
+const LoadingProvider = (props) => {
+  const { children } = props;
+  const [loading, setLoading] = useState(false);
+  const show = () => setLoading(true);
+  const hide = () => setLoading(false);
+  const valueContext = useMemo(() => ({
+    loading,
+    show,
+    hide,
+  }), [loading, show, hide]);
```

Comment:
Please explain this.

Reply:
function show / hide will reset the value to true / false used to display the loading state

Reply:
Why do you use `show` and `hide` as dependencies?

Reply:
because it only changes the state of the loading

Reply:
Why does it change the loading state?

Reply:
I have a mistake here on how to use it, I will review and correct it